### PR TITLE
Update dependent tab to include clinical documents for AB#14367

### PR DESCRIPTION
**Pull request by BrianMaki**
# Implements [AB#14367](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/14367)

## Description

Also implements [AB#14368](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/14368)

- Added clinical docs tab in dependents
- Added snow plow call for download clinical document
- Added download clinical doc report button in table
- Added configuration for clinical docs tab in app settings.
- Added functional tests for dependent clinical docs
- Cleaned up functional tests to use string interpolation and comments

## Testing

- [ ] Unit Tests Updated
- [x] Functional Tests Updated
- [ ] Not Required

## UI Changes

**Clinical Docs PDF:**

Screenshot 2022-12-01 at 9 35 19 AM

**Clinical Docs Tab Desktop:**

Screenshot 2022-11-23 at 9 05 19 PM

**Clinical Docs Tab Mobile**

Screenshot 2022-11-23 at 9 09 06 PM

**Clinical Docs Download:**

Screenshot 2022-11-23 at 9 11 29 PM

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

**Comment by mk08sh** [comment]
👍

**Comment by mk08sh** [comment]
👍

**Comment by ninosamson** [comment]
looks good!

**Review by rayking29 — APPROVED** [review]

File: Apps/WebClient/src/ClientApp/src/components/DependentCardComponent.vue [hunk 1]
```diff
@@ -1341,6 +1447,119 @@ export default class DependentCardComponent extends Vue {
...
+                            </b-tr>
+                        </b-thead>
+                        <b-tbody>
+                            <b-tr
+                                v-for="(row, index) in clinicalDocuments"
+                                :key="index"
+                            >
+                                <b-td
+                                    :data-testid="`clinical-document-service-date-${dependent.ownerId}-${index}`"
+                                    class="align-middle"
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/src/components/DependentCardComponent.vue:1512]: Should add `text-nowrap` class per #4466.

File: Testing/functional/tests/cypress/integration/e2e/user/dependents.js [hunk 2]
```diff
@@ -310,18 +305,14 @@ describe("dependents", () => {
...
         ).click({ force: true });
 
-        // Click XLSX
+        // Click CSV
         cy.get(
-            "[data-testid=download-immunization-forecast-report-xlsx-btn-" +
-                validDependentHdid +
-                "]"
+            `[data-testid=download-immunization-forecast-report-xlsx-btn-${validDependentHdid}]`
         ).click({ force: true });
```
Comment (MikeLyttle) [inline Testing/functional/tests/cypress/integration/e2e/user/dependents.js:316]: Seems like this is clicking XLSX button, not CSV.  Typo in comment?

File: Apps/WebClient/src/ClientApp/src/components/DependentCardComponent.vue [hunk 3]
```diff
@@ -1341,6 +1447,119 @@ export default class DependentCardComponent extends Vue {
...
+                <b-tab
+                    v-if="isClinicalDocumentTabShown"
+                    :disabled="isExpired"
+                    no-body
+                    :data-testid="`clinical-document-tab-${dependent.ownerId}`"
+                    @click="fetchClinicalDocuments"
+                >
+                    <template #title>
+                        <div
+                            :data-testid="`clinical-docuemnt-tab-title-${dependent.ownerId}`"
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/src/components/DependentCardComponent.vue:1459]: Typo: docuemnt should be document.  (Applies to references in functional tests as well.)

**Review by MikeLyttle — CHANGES_REQUESTED** [review]

File: Apps/WebClient/src/ClientApp/src/components/DependentCardComponent.vue [hunk 4]
```diff
@@ -898,7 +1003,12 @@ export default class DependentCardComponent extends Vue {
                         borderless
                         :items="testRows"
                         class="w-100 mb-0"
+<<<<<<< HEAD
+                        aria-describedby="COVID-19 Test Results"
+                        :data-testid="`covid19-table-${dependent.ownerId}`"
+=======
                         aria-describedby="COVID‑19 Test Results"
+>>>>>>> 13647
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/src/components/DependentCardComponent.vue:1011]: There are some merge conflict indicators showing up.

Comment (BrianMaki) [reply Apps/WebClient/src/ClientApp/src/components/DependentCardComponent.vue]: Fixed

File: Apps/WebClient/src/ClientApp/src/components/DependentCardComponent.vue (hunk 3 above)
Comment (BrianMaki) [reply Apps/WebClient/src/ClientApp/src/components/DependentCardComponent.vue]: Fixed

File: Testing/functional/tests/cypress/integration/e2e/user/dependents.js (hunk 2 above)
Comment (BrianMaki) [reply Testing/functional/tests/cypress/integration/e2e/user/dependents.js]: Fixed

File: Apps/WebClient/src/ClientApp/src/components/DependentCardComponent.vue (hunk 1 above)
Comment (BrianMaki) [reply Apps/WebClient/src/ClientApp/src/components/DependentCardComponent.vue]: Fixed

**Review by MikeLyttle — APPROVED** [review]

**Pull request merged**